src/variable_info.hpp

```diff
@@ -16,98 +16,119 @@
 
 #pragma once
 
-/** Information on a WML variable. */
-#include <string>
 #include "config.hpp"
 #include "variable_info_detail.hpp"
 
-class invalid_variablename_exception : public std::exception
+#include <string>
+#include <type_traits>
+
+// Restrict alias to local scope.
+namespace
```

Comment:
What the heck are you doing here? This is a header. Why do you want an anonymous namespace? What good will it do?